service/geopmdpy/restorable_file_writer.py

```diff
@@ -56,6 +56,8 @@ def backup_and_try_update(self, new_file_contents: str):
             # contents, since that setting may be what the service wrote before.
             self._warning_handler(f'Reusing existing backup at {self._backup_path}')
         else:
+            system_files.secure_make_dirs(os.path.dirname(self._backup_path),
+                                          perm_mode=system_files.GEOPM_SERVICE_RUN_PATH_PERM)
```

Comment:
Why not put this in main?

Reply:
```
cmcantal@ubuntu22vm:~/Git/geopm/service/geopmdpy (issue-3161)]$ git diff
diff --git a/service/geopmdpy/__main__.py b/service/geopmdpy/__main__.py
index 80231bfa7..8aabb286d 100644
--- a/service/geopmdpy/__main__.py
+++ b/service/geopmdpy/__main__.py
@@ -34,6 +34,8 @@ def stop():
 def main():
     signal(SIGTERM, term_handler)
     global _bus, _loop
+    system_files.secure_make_dirs(system_files.GEOPM_SERVICE_RUN_PATH,
+                                  perm_mode=system_files.GEOPM_SERVICE_RUN_PATH_PERM)
     _loop = EventLoop()
     _bus = SystemMessageBus()
     with RestorableFileWriter(
diff --git a/service/geopmdpy/restorable_file_writer.py b/service/geopmdpy/restorable_file_writer.py
index 9224ae99d..480d8a8c0 100644
--- a/service/geopmdpy/restorable_file_writer.py
+++ b/service/geopmdpy/restorable_file_writer.py
@@ -56,8 +56,6 @@ class RestorableFileWriter:
             # contents, since that setting may be what the service wrote before.
             self._warning_handler(f'Reusing existing backup at {self._backup_path}')
         else:
-            system_files.secure_make_dirs(os.path.dirname(self._backup_path),
-                                          perm_mode=system_files.GEOPM_SERVICE_RUN_PATH_PERM)
             system_files.secure_make_file(self._backup_path, old_setting)
 
         with open(self._write_path, 'w') as f:
```